File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int sizeOfLL(ListNode head){
        ListNode current=head;
        int size=0;
        while(current!=null){
            size++;
            current=current.next;
        }
        return size;
    }
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int size=sizeOfLL(head);
        int target=size-n+1;

        if(target==1){
            return head.next;
        }
        int i=1;
        ListNode current=head;

        while(current!=null && i<target-1){
            current=current.next;
            i++;
        }
        current.next=current.next.next;
        return head;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int sizeOfLL(ListNode* head){
        int len=0;
        ListNode* temp=head;
        while(temp!=NULL){
            len++;
            temp=temp->next;
        }
        return len;
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int position=sizeOfLL(head)-n+1;

        if(position==1){
            ListNode* toDelete=head;
            head=head->next;
            delete toDelete;
            return head;
        }

        // else 
        ListNode* previous=NULL;
        ListNode* current=head;
        int Cposition=1;
        while(Cposition<position){
            ListNode* forward=current->next;
            previous=current;
            current=forward;
            Cposition++;
        }
        previous->next=current->next;
        delete current;
        return head;
    }
};